### PR TITLE
Preference center via API guide and endpoint docs EA

### PR DESCRIPTION
**Pull request by lydia-xie**
# Pull Request/Issue Resolution

Closes #**ISSUE_NUMBER_HERE**

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [ ] No

---

### CODE REVIEW

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+|---|---|
+| Enabled preference center | Your Braze dashboard has permissions to use the preference center feature. |
+| Valid app group with an email subscription group | A working app group with valid users and an email subscription group. |
+| Valid user | A user with an email address and an external ID. |
+| Generated API key with preference center permissions | In the Braze dashboard, go to **Developer Console** > **API Settings** to confirm that you have access to an API key with preference center permissions. |
+{: .reset-td-br-1 .reset-td-br-2}
+
+## Step 1: Create a preference center via API
+
+Let's begin building a preference center using the [`/preference_center/v1/{preferenceCenterExternalId}/url/{userId}`]({{site.baseurl}}/api/endpoints/preference_center/post_create_preference_center/) endpoint. To customize your preference center, you can include HTML that aligns with your branding for the `preference_center_page_html` field and `confirmation_page_html` field.
```
Comment (tiffanyjlin) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:30]: should be `/preference_center/v1` to create, and `/preference_center/v1/{preferenceCenterExternalId}` to update

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Step 2: Create a preference center URL
```
Comment (tiffanyjlin) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:32]: We probably can skip this step, as the user can include the liquid tag `{{preference_center.${preference_center_name}}}}` directly in an email, and the url will generate when the email is sent from Braze.

The `/preference_center/v1/{preferenceCenterExternalId}/url/{userId}` endpoint allows customers to grab the URL for a specific user outside of an email that is sent through Braze.

File: _docs/_api/endpoints/preference_center/post_create_preference_center.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+```
+
+## Request parameters
+
+| Parameter | Required | Data Type | Description |
+| --------- | ---------| --------- | ----------- |
+|`name`| Required | String | The name of the preference center. |
+|`preference_center_page_html`| Optional | String | The HTML for the preference center page. |
+|`confirmation_page_html`| Optional | String | The HTML for the confirmation page. |
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/post_create_preference_center.md:48]: The `preference_center_page_html` and `confirmation_page_html `  parameters are required.

File: _docs/_api/endpoints/preference_center/get_view_details_preference_center.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+--header 'Authorization: Bearer YOUR-REST-API-KEY'
+```
+
+## Response 
+```json 
+Content-Type: application/json
+Authorization: Bearer YOUR-REST-API-KEY
+{
+  "name": "My Preference Center",
+  "preference_center_api_id": "preference_center_external_id",
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_view_details_preference_center.md:40]: I think that we should stick with the parlance `preference_center_api_id`. i.e. replace all instances of `preference_center_external_id` with `preference_center_api_id`. After all, from a user's prospective, this ID isn't external to anything.

File: _docs/_api/endpoints/preference_center/get_view_details_preference_center.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+curl --location -g --request GET https://rest.iad-01.braze.com/preference_center/v1/preference_center_external_id \
+--header 'Authorization: Bearer YOUR-REST-API-KEY'
+```
+
+## Response 
+```json 
+Content-Type: application/json
+Authorization: Bearer YOUR-REST-API-KEY
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_view_details_preference_center.md:37]: The response would not include the API key.

File: _docs/_api/endpoints/preference_center/get_list_preference_center.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+curl --location -g --request GET https://rest.iad-01.braze.com/preference_center/v1/list \
+--header 'Authorization: Bearer YOUR-REST-API-KEY'
+```
+
+## Response
+```json
+Content-Type: application/json
+Authorization: Bearer YOUR-REST-API-KEY
+{
+  "preference_centers":
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_list_preference_center.md:39]: The JSON object should be contained within a JSON array.  So it should be something like 
{ "preference_centers": [ { "name": "Example", ....

File: _docs/_api/endpoints/preference_center/get_create_url_preference_center.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+|`name`| Required | String | The name of the preference center. |
+|`preference_center_page_html`| Required | String | The HTML for the preference center page. |
+|`confirmation_page_html`| Required | String | The HTML for the confirmation page. |
+{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3  .reset-td-br-4}
+
+## Response 
+
+```json
+Content-Type: application/json
+Authorization: Bearer YOUR-REST-API-KEY
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_create_url_preference_center.md:47]: Response would not contain the api key.

File: _docs/_api/endpoints/preference_center/get_create_url_preference_center.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+--header 'Authorization: Bearer YOUR-API-KEY-HERE'
+```
+
+## Request parameters
+
+| Parameter | Required | Data Type | Description |
+| --------- | ---------| --------- | ----------- |
+|`name`| Required | String | The name of the preference center. |
+|`preference_center_page_html`| Required | String | The HTML for the preference center page. |
+|`confirmation_page_html`| Required | String | The HTML for the confirmation page. |
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_create_url_preference_center.md:40]: This looks like it was copied from the create endpoint. The URL parameters for this endpoint are the preference-center-api-id and the user's external-id.

File: _docs/_api/endpoints/preference_center/get_view_details_preference_center.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+article_title: "GET: View Details for Preference Centers"
+search_tag: Endpoint
+page_order: 3
+layout: api_page
+page_type: reference
+description: "This article outlines details about the View details for preference centers Braze endpoint."
+
+---
+{% api %}
+# List preference centers
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_view_details_preference_center.md:12]:
``` suggestion
View Details for Preference Center
```

File: _docs/_api/endpoints/preference_center/get_view_details_preference_center.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+---
+nav_title: "GET: View Details for Preference Centers"
+article_title: "GET: View Details for Preference Centers"
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/get_view_details_preference_center.md:3]:
``` suggestion
article_title: "GET: View Details for Preference Center"
```

File: _docs/_api/endpoints/preference_center/post_create_preference_center.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  "confirmation_page_html":"<html>
+    <head>
+    </head>
+    <body>Thanks for updating your preferences!
+      </body>
+</html>",
+}'
+```
+{% endraw %}
+
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/post_create_preference_center.md:101]: A sample response body should be show somewhere here.

File: _docs/_api/endpoints/preference_center/put_update_preference_center.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+  "preference_center_page_html": "string",
+  "confirmation_page_html": "string"
+}
+```
+
+## Request parameters
+
+| Parameter | Required | Data Type | Description |
+| --------- | ---------| --------- | ----------- |
+|`name`| Required | String | The name of the preference center. |
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/put_update_preference_center.md:46]: `name` is not a valid parameter. The preference center names cannot be changed.

File: _docs/_api/endpoints/preference_center/put_update_preference_center.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+|`preference_center_page_html`| Required | String | The HTML for the preference center page. |
+|`confirmation_page_html`| Required | String | The HTML for the confirmation page. |
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/put_update_preference_center.md:48]: Only one of the parameters `preference_center_page_html`, `confirmation_page_html` or `redirect_page_html` is required.

File: _docs/_api/endpoints/preference_center/put_update_preference_center.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+--header 'Content-Type: application/json' \
+--header 'Authorization: Bearer YOUR-API-KEY-HERE' \
+--data-raw '{
+  "name": "Example",
+  "preference_center_page_html": "HTML for preference center here"
+  "confirmation_page_html": "HTML here with a message to users here"
+
+```
+{% endraw %}
+
```
Comment (jteich) [inline _docs/_api/endpoints/preference_center/put_update_preference_center.md:64]: should show sample response body somewhere here.

File: _docs/_api/endpoints/templates/email_templates/post_update_email_template.md [hunk 15]
```diff
@@ -14,7 +14,7 @@ description: "This article outlines details about the Update Email Template Braz
 /templates/email/update
 {% endapimethod %}
 
-Use this endpoint to updates email templates on the Braze dashboard. You can access an email template's `email_template_id` by navigating to it on the **Templates & Media** page. The email template creation API endpoint will also return an `email_template_id` reference.
+Use this endpoint to update email templates on the Braze dashboard. You can access an email template's `email_template_id` by navigating to it on the **Templates & Media** page. The [Create email template endpoint]({{site.baseurl}}/api/endpoints/templates/email_templates/post_create_email_template/) will also return an `email_template_id` reference.
```
Comment (jteich) [inline _docs/_api/endpoints/templates/email_templates/post_update_email_template.md:18]: this looks out of scope for preference center.

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+The Braze endpoints used to create a preference center are currently in early access. Contact your Braze account manager if you're interested in participating in the early access.
+{% endalert %}
+
+## Prerequisites
+
+| Requirement | Description |
+|---|---|
+| Enabled preference center | Your Braze dashboard has permissions to use the preference center feature. |
+| Valid app group with an email subscription group | A working app group with valid users and an email subscription group. |
+| Valid user | A user with an email address and an external ID. |
```
Comment (jteich) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:24]: truly speaking, the user doesn't need an email address. In its current form, it wouldn't be of much use to a user without an email address since it's only used to set email preferences. However, in future iterations, this will be available for other channels as well.

File: _docs/_api/endpoints/templates/email_templates/post_update_email_template.md (hunk 15 above)
Comment (lydia-xie) [reply _docs/_api/endpoints/templates/email_templates/post_update_email_template.md:18]: yep, this was just an edit I included in this PR!

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+{: .reset-td-br-1 .reset-td-br-2}
+
+## Step 1: Create a preference center via API
+
+{% raw %}
+Let's begin building a preference center using the [`/preference_center/v1`]({{site.baseurl}}/api/endpoints/preference_center/post_create_preference_center/) endpoint. To customize your preference center, you can include HTML that aligns with your branding for the `preference_center_page_html` field and `confirmation_page_html` field.
+
+The [`/preference_center/v1/{preferenceCenterExternalId}/url/{userId}`]({{site.baseurl}}/api/endpoints/preference_center/get_create_url_preference_center/) endpoint allows you to grab the preference center URL for a specific user outside of an email that is sent through Braze.
+
+## Step 3: Include in email campaign
```
Comment (tiffanyjlin) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:35]: should now be Step 2 since we deleted a step

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+
+Next, include insert your preference center into your email campaign by using a combination of HTML that includes Liquid. Here's an example of what you could include at an email footer to indicate to your users that they can edit their preferences.
```
Comment (tiffanyjlin) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:37]: instead of "using a combination of HTML that includes Liquid" maybe something like: "pasting the Liquid tag for that preference center in the URL of a link"

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+```
+{%endraw%}
+
+## Editing a preference center
+
+You can edit and update your preference center by using the [`/preference_center/v1/{preferenceCenterExternalId}`]({{site.baseurl}}/api/endpoints/preference_center/put_update_preference_center/) endpoint. 
+
+## Identifying preference centers and details
+
+To identify your preference centers, use the [`/preference_center/v1/{preferenceCenterExternalId}`]({{site.baseurl}}/api/endpoints/preference_center/get_view_details_preference_center/) endpoint to return related information such as the a last updated timestamp, the preference center ID, and more.
```
Comment (tiffanyjlin) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:50]: remove "a" from "the a last updated"

File: _docs/_api/endpoints/preference_center/get_create_url_preference_center.md [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+description: "This article outlines details about the Generate preference center URL Braze endpoint."
+
+---
+{% api %}
+# Generate preference center URL
+{% apimethod get %}
+/preference_center/v1/{preferenceCenterExternalId}/url/{userId}
+{% endapimethod %}
+
+Use this endpoint to generate a URL for a preference center.
```
Comment (tiffanyjlin) [inline _docs/_api/endpoints/preference_center/get_create_url_preference_center.md:17]: can we add that it's a URL for a preference center for a specific user?

File: _docs/_api/endpoints/preference_center/post_create_preference_center.md [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+## Request example
+{% raw %}
+```
+curl --location --request POST 'https://rest.iad-01.braze.com/preference_center/v1' \
+--header 'Content-Type: application/json' \
+--header 'Authorization: Bearer YOUR-API-KEY-HERE' \
+--data-raw '{
+  "name": "Example Preference Center",
+  "preference_center_page_html": "<html>
+  <head></head>
```
Comment (tiffanyjlin) [inline _docs/_api/endpoints/preference_center/post_create_preference_center.md:60]: Can we remove this example for now? I'd like to replace with a formatted/styled HTML example in the future (to be provided)

File: _docs/_api/endpoints/preference_center/post_create_preference_center.md [hunk 22]
```diff
@@ -0,0 +1,114 @@
...
+
+## Request parameters
+
+| Parameter | Required | Data Type | Description |
+| --------- | ---------| --------- | ----------- |
+|`name`| Required | String | The name of the preference center. |
+|`preference_center_page_html`| Required | String | The HTML for the preference center page. |
+|`confirmation_page_html`| Required | String | The HTML for the confirmation page. |
+{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3  .reset-td-br-4}
+
```
Comment (tiffanyjlin) [inline _docs/_api/endpoints/preference_center/post_create_preference_center.md:50]: can we add a section about each of the new liquid tags to be included in the HTML? Feel free to wordsmith to make it easier to understand

Liquid tags for generating user's subscribed state on the page:

- {{subscribed_state.${email_global}}}: get the global email subscribed state for the user (i.e. "opted_in" "subscribed" or "unsubscribed"
- {{subscribed_state.${<subscription-group-id>}}}: get the subscribed state of a subscription group for the user (i.e. "subscribed" or "unsubscribed")

Liquid tags for the form input elements:
- {% form_field_name :email_global_state %}: indicates that a specific form input element corresponds to the user's global email susbcribed state. User's selection state should be "opted_in", "subscribed", or "unsubscribed" when the form is submitted with selection data for the global email subscribed state. If it's a checkbox, the user will either be "opted_in" or "unsubscribed". If it's a hidden input, the "subscribed" state will also be valid. 
- {% form_field_name :subscription_group <subscription-group-id> %}: indicates that a specific form input element corresponds to a given subscription group. User's selection state should be either "subscribed" or "unsubscribed" when the form is submitted with selection data for a specific subscription group.

Liquid tag for form action:
- {{preference_center_submit_url}}: generates URL for form submission

File: _docs/_api/endpoints/preference_center/put_update_preference_center.md [hunk 23]
```diff
@@ -0,0 +1,76 @@
+---
+nav_title: "PUT: Update Preference Center"
+article_title: "PUT: Update Preference Center"
+search_tag: Endpoint
+page_order: 5
+layout: api_page
+page_type: reference
+description: "This article outlines details about the Update a preference center Braze endpoint."
```
Comment (tiffanyjlin) [inline _docs/_api/endpoints/preference_center/put_update_preference_center.md:8]: should we link to the Create Pref Center page here, since the details are the same?

File: _docs/_api/endpoints/preference_center/post_create_preference_center.md [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+Refer to the following Liquid tags that can be included in your HTML to generate a user's subscription state on the preference center page.
+
+{% raw %}
+
+#### User subscription state
+
+| Liquid | Description |
+| --------- | ---------|
+|`{{subscribed_state.${email_global}}}`| Get the global email subscribed state for the user (i.e. "opted_in" "subscribed" or "unsubscribed". |
+|`{{subscribed_state.${}}}`| Get the subscribed state of a subscription group for the user (i.e. "subscribed" or "unsubscribed"). |
```
Comment (tiffanyjlin) [inline _docs/_api/endpoints/preference_center/post_create_preference_center.md:62]: Can we specify that the user should copy/paste the Subscription Group ID into the liquid tag? {{subscribed_state.${<subscription group id here>}}}

File: _docs/_user_guide/message_building_by_channel/email/preference_center.md [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+
+## Editing a preference center
+
+You can edit and update your preference center by using the [`/preference_center/v1/{preferenceCenterExternalId}`]({{site.baseurl}}/api/endpoints/preference_center/put_update_preference_center/) endpoint. 
+
+## Identifying preference centers and details
+
+To identify your preference centers, use the [`/preference_center/v1/{preferenceCenterExternalId}`]({{site.baseurl}}/api/endpoints/preference_center/get_view_details_preference_center/) endpoint to return related information such as the last updated timestamp, the preference center ID, and more.
+
+`
```
Comment (KellieHawks) [inline _docs/_user_guide/message_building_by_channel/email/preference_center.md:52]: Can you remove the "`". Don't want to mess around with this version while you are making edits.

**Review by KellieHawks — APPROVED** [review]
